Mqtt add object

Smeagolworms4 opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

Add documentation of object_id in relation with with PR: https://github.com/home-assistant/core/pull/58728

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase:  https://github.com/home-assistant/core/pull/58728
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

frenck committed:
2021.11: Initial beta release notes

frenck committed:
Merge branch 'next' into rc

frenck committed:
Sync codebase with docs for 2021.11

rytilahti committed:
Document wled number platform (#20039)

Smeagolworms4 committed:
Add MQTT object_id option

emontnemery commented:
Each MQTT entity's documentation should be updated with the new configuration parameter, not just the discovery starting page.

emontnemery reviewed (changes requested):
Each MQTT entity's documentation should be updated with the new configuration parameter, not just the discovery starting page.

emontnemery committed:
Revert "Add documentation for supplying offset to time trigger" (#20119)

frenck committed:
Merge branch 'current' into rc

frenck committed:
Merge branch 'current' into rc

frenck committed:
2021.11: Update & sort breaking changes section

frenck committed:
2021.11: Update changelog

frenck committed:
2021.11: Add daylight saving time bug section

frenck committed:
2021.11: Visit device & service section

frenck committed:
2021.11: Tuya update

frenck committed:
2021.11: Tweaks

frenck committed:
2021.11: Intro & TOC

frenck committed:
2021.11: Icon picker section

frenck committed:
2021.11: Entity category section

frenck committed:
2021.11: Tweaks

frenck committed:
2021.11: Template section

frenck committed:
2021.11: Other noteworthy changes

frenck committed:
2021.11: Tweaks

frenck committed:
2021.11: Review comments

frenck committed:
2021.11: Review comments

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

frenck committed:
2021.11: Latest changelog

frenck committed:
Merge pull request #20137 from home-assistant/rc

frenck committed:
2021.11: Fix missing Vicare breaking change

OttoWinter committed:
Fix typo in 2021.11 release notes (#20140)

Chupaka committed:
Fix a typo (#20139)

emontnemery committed:
Update recorder documentation according to ADR18 (#20145)

Chupaka committed:
its != it's 😊 (#20141)

emontnemery committed:
Adjust minimum supported version of SQLite (#20152)

balloob committed:
2021.11.1

MartinHjelmare committed:
Remove migration section from ozw (#20162)

jbouwh commented on source/_docs/mqtt/discovery.markdown:
```diff
@@ -481,3 +482,19 @@ If the device supports gps coordinates then they can be sent to Home Assistant b
+### Used object_id to generate the entity id
+
+The entity identifier is generated from the device name. If you defined an object_id then it will be used instead.
+- Configuration topic: `homeassistant/sensor/devive1/config`
```
Spelling device

jbouwh commented on source/_docs/mqtt/discovery.markdown:
```diff
@@ -481,3 +482,19 @@ If the device supports gps coordinates then they can be sent to Home Assistant b
+  "name":"My Super Device",
+  "object_id":"device1",
+  "state_topic": "homeassistant/sensor/devive1/state",
+ }
```
Same here

nick-seward committed:
docs: Adding to the Slack integration examples (#20172)

* docs: Adding to the Slack integration examples

I'm proposing two new examples on how to message someone directly in Slack and how to mention (ex. @user) a user in a Slack channel.

* doc: Fixing spelling mistake

slovdahl committed:
Add link to docs for condition trigger ID (#20169)

Smeagolworms4 committed:
add object_id doc each entities type

Smeagolworms4 committed:
Spelling
